### PR TITLE
Add some fixes to foreman release procedure

### PR DESCRIPTION
**Pull request by zjhuntin**

### CODE REVIEW

File: procedures/foreman/release.md.erb [hunk 1]
```diff
@@ -49,13 +51,15 @@ Note: If for some reason there was an issue with the tarballs that required uplo
 # Packaging a release
 
 [Background documentation](https://projects.theforeman.org/projects/foreman/wiki/Release_Process#Packaging-a-release)
-
+* [ ] Make sure all nightly builds of packages are removed from the releases tags
```
Comment (ekohl) [inline procedures/foreman/release.md.erb]: Usually we don't build anything nightly in release tags. Did you intend this in the branch docs?

File: procedures/foreman/release.md.erb [hunk 2]
```diff
@@ -49,13 +51,15 @@ Note: If for some reason there was an issue with the tarballs that required uplo
...
 <% if is_rc %>* [ ] [Download](https://github.com/theforeman/theforeman-rel-eng/blob/master/download_extras), [sign](https://github.com/theforeman/theforeman-rel-eng/blob/master/sign_extras) and [upload](https://github.com/theforeman/theforeman-rel-eng/blob/master/upload_extras) "extra" RPMs
 <% end %>* [ ] Kick off the [release pipeline](https://ci.theforeman.org/job/foreman-<%= short_version %>-release-pipeline/) by calling [`release_pipeline`](https://github.com/theforeman/theforeman-rel-eng/blob/master/release_pipeline)
+* [ ] With your project as Client [Download](https://github.com/theforeman/theforeman-rel-eng/blob/master/download_rpms), [sign](https://github.com/theforeman/theforeman-rel-eng/blob/master/sign_rpms), [upload RPM signatures](https://github.com/theforeman/theforeman-rel-eng/blob/master/upload_rpm_signatures) and [upload RPMs](https://github.com/theforeman/theforeman-rel-eng/blob/master/upload_rpms)
```
Comment (ekohl) [inline procedures/foreman/release.md.erb]: 3 lines above is the same instruction

File: procedures/foreman/release.md.erb [hunk 3]
```diff
@@ -49,13 +51,15 @@ Note: If for some reason there was an issue with the tarballs that required uplo
 # Packaging a release
 
 [Background documentation](https://projects.theforeman.org/projects/foreman/wiki/Release_Process#Packaging-a-release)
-
+* [ ] Make sure all nightly builds of packages are removed from the releases tags
 * [ ] Update [foreman-packaging](https://github.com/theforeman/foreman-packaging) [rpm/<%= short_version %>](https://github.com/theforeman/theforeman-rel-eng/blob/master/bump_rpm_packaging) and [deb/<%= short_version %>](https://github.com/theforeman/theforeman-rel-eng/blob/master/bump_deb_packaging) branches
 * [ ] Trigger [release_packages in Jenkins](https://ci.theforeman.org/job/release_packages/) by calling [release_packages](https://github.com/theforeman/theforeman-rel-eng/blob/master/release_packages)
 * [ ] Tag `foreman-release` into `foreman-client-<%= short_version %>-rhel7`, as this doesn't happen automatically for that one tag: `koji tag-build foreman-client-<%= short_version %>-rhel7 foreman-release-<%= version %>-1.el7`<% if is_rc %> (make sure the build matches the release)<% end %>
+* [ ] Make sure the [Nightly plugins pipeline](https://ci.theforeman.org/job/foreman-plugins-nightly-rpm-pipeline) has passed recently
```
Comment (ekohl) [inline procedures/foreman/release.md.erb]: Normally nightly is not important for a release

File: procedures/foreman/release.md.erb (hunk 1 above)
Comment (zjhuntin) [reply procedures/foreman/release.md.erb]: Yes I remember that there was a point where I needed to do this, just got the location in the steps wrong.

File: procedures/foreman/release.md.erb (hunk 2 above)
Comment (zjhuntin) [reply procedures/foreman/release.md.erb]: derp, I think Evgeni added this earlier than I did. Will remove!

File: procedures/foreman/release.md.erb (hunk 3 above)
Comment (ehelms) [reply procedures/foreman/release.md.erb]: I think this is important for branching process so we branch a working codebase.

Comment (ekohl) [reply procedures/foreman/release.md.erb]: Correct, but that's `branch.md.erb`, not `release.md.erb`.

Comment (zjhuntin) [reply procedures/foreman/release.md.erb]: As I recall, there was something that failed for me if this hadn't passed recently... Lemme see if I can find the part.

Comment (zjhuntin) [reply procedures/foreman/release.md.erb]: Hrm, not sure what was blocking me but wasn't this. Will remove and if I recall what it is I'll reupdate.

**Comment by ekohl** [comment]
I'll leave this open in case @evgeni or @ehelms have another comment but LGTM.

File: procedures/foreman/branch.md.erb [hunk 4]
```diff
@@ -39,14 +41,15 @@
 * [ ] [Generate](https://github.com/theforeman/theforeman-rel-eng/blob/master/generate_gpg), [backup](https://github.com/theforeman/theforeman-rel-eng/blob/master/export_gpg_private), [sign](https://github.com/theforeman/theforeman-rel-eng/blob/master/sign_gpg) and [upload](https://github.com/theforeman/theforeman-rel-eng/blob/master/upload_gpg) a GPG key. See [GPG_Keys](https://projects.theforeman.org/projects/foreman/wiki/GPG_Keys) if needed.
 * [ ] Publish the key by [exporting](https://github.com/theforeman/theforeman-rel-eng/blob/master/export_gpg_public) the GPG key
   * [ ] Update the website's [security.md](https://github.com/theforeman/theforeman.org/blob/gh-pages/security.md) and create a file in [static/keys](https://github.com/theforeman/theforeman.org/tree/gh-pages/static/keys)
-  * [ ] Create [releases/<%= release %>/RPM-GPG-KEY-foreman](https://yum.theforeman.org/releases/<%= release %>/RPM-GPG-KEY-foreman) on yum.theforeman.org
```
Comment (evgeni) [inline procedures/foreman/branch.md.erb]: Nit: there is a slash missing in front of var and the whole path could go in `backticks` for better recognition.

**Pull request merged**